modin/pandas/test/dataframe/test_protocol.py

```diff
@@ -0,0 +1,274 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+import modin.pandas as pd
+import numpy as np
+import pytest
+from modin.pandas.test.utils import df_equals
+from modin.pandas.utils import from_dataframe
+from modin.core.dataframe.base.exchange.dataframe_protocol.utils import (
+    DTypeKind,
+    ColumnNullType,
+)
+from modin.config import StorageFormat
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        {"a": [1.5, 2.5, 3.5], "b": [9.2, 10.5, 11.8]},
+        {"a": [1, 2, 3], "b": [3, 4, 5], "c": [1.5, 2.5, 3.5], "d": [9, 10, 11]},
+    ],
+)
+def test_float_only(data):
+    df = pd.DataFrame(data)
+    df_equals(df, from_dataframe(df.__dataframe__()))
+
+
+def test_noncontiguous_columns():
+    arr = np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+    df = pd.DataFrame(arr, columns=["a", "b", "c"])
+    if os.name == "nt":
+        assert df["a"].to_numpy().strides == (4,)
+    elif os.name == "posix":
+        assert df["a"].to_numpy().strides == (8,)
```

Comment:
[link to the original discission](https://github.com/modin-project/modin/pull/4265#discussion_r824142171)

Reply:
@dchigarev I removed this test